channel/src/main/java/com/synopsys/integration/alert/channel/msteams2/MSTeamsChannelMessageSender.java

```diff
@@ -0,0 +1,104 @@
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.msteams2;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.synopsys.integration.alert.channel.api.ChannelMessageSender;
+import com.synopsys.integration.alert.channel.util.RestChannelUtility;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+import com.synopsys.integration.alert.common.message.model.MessageResult;
+import com.synopsys.integration.alert.common.persistence.model.job.details.MSTeamsJobDetailsModel;
+import com.synopsys.integration.alert.descriptor.api.MsTeamsKey;
+import com.synopsys.integration.alert.processor.api.extract.model.ProviderDetails;
+import com.synopsys.integration.rest.request.Request;
+
+@Component
+public class MSTeamsChannelMessageSender implements ChannelMessageSender<MSTeamsJobDetailsModel, MSTeamsChannelMessageModel, MessageResult> {
+    private static final String MESSAGE_THEME_COLOR = "5A2A82"; // Synopsys Purple
+    private static final String MESSAGE_SUMMARY = "New Content from Alert";
+
+    private final RestChannelUtility restChannelUtility;
+    private final MsTeamsKey msTeamsKey;
+
+    @Autowired
+    public MSTeamsChannelMessageSender(RestChannelUtility restChannelUtility, MsTeamsKey msTeamsKey) {
+        this.restChannelUtility = restChannelUtility;
+        this.msTeamsKey = msTeamsKey;
+    }
+
+    @Override
+    public MessageResult sendMessages(MSTeamsJobDetailsModel msTeamsJobDetailsModel, List<MSTeamsChannelMessageModel> channelMessages) throws AlertException {
+        String webhook = msTeamsJobDetailsModel.getWebhook();
+
+        Map<String, String> requestHeaders = new HashMap<>();
+        requestHeaders.put("Content-Type", "application/json");
+
+        String provider = channelMessages.stream()
+                              .findFirst()
+                              .map(MSTeamsChannelMessageModel::getProviderDetails)
+                              .map(ProviderDetails::getProvider)
+                              .map(LinkableItem::getValue)
+                              .orElse("Black Duck"); // TODO: Should we always assume we have at least one channel message to avoid this Optional?
```

Comment:
I would put it in a private method for readability, but we can remove the TODO. I think it's fine to provide a default. Also, I think we are supposed to have one word for "BlackDuck".